ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt

```diff
@@ -125,26 +125,23 @@ constructor(
    * Attempts to upload a single photo to remote storage. Returns an [Result] indicating whether the
    * upload attempt failed or succeeded.
    */
-  private suspend fun uploadPhotoMedia(photoTaskData: PhotoTaskData): kotlin.Result<Unit> {
-    val path = photoTaskData.remoteFilename
-    val photoFile = userMediaRepository.getLocalFileFromRemotePath(path)
-    if (!photoFile.exists()) {
-      Timber.e("Photo not found. local path: ${photoFile.path}, remote path: $path")
-      return kotlin.Result.failure(
-        FileNotFoundException("Photo $path not found on device: ${photoFile.path}")
-      )
-    }
-
-    Timber.d("Starting photo upload. local path: ${photoFile.path}, remote path: $path")
-    return try {
+  private suspend fun uploadPhotoMedia(photoTaskData: PhotoTaskData): kotlin.Result<Unit> =
+    try {
+      val path = photoTaskData.remoteFilename
+      val photoFile = userMediaRepository.getLocalFileFromRemotePath(path)
+      Timber.d("Starting photo upload. local path: ${photoFile.path}, remote path: $path")
+      if (!photoFile.exists()) {
+        throw FileNotFoundException(photoFile.path)
+      }
       remoteStorageManager.uploadMediaFromFile(photoFile, path)
       kotlin.Result.success(Unit)
+    } catch (t: FirebaseException) {
+      Timber.d(t, "Can't connect to Firebase to upload photo")
```

Comment:
Can we change the log level to `Timber.e`, since it could be any exception (and not just connectivity issue)?

Reply:
Crashlytics logs were noisy due to users being offline. How about I update this to only log the more specific `FirebaseNetworkException` as debug, all others as error?